config/config_test.go

```diff
@@ -291,23 +291,40 @@ func TestLoad_tracing(t *testing.T) {
 
 func TestCustomCertsDataDecoder(t *testing.T) {
 	var c Config
-	os.Setenv("TYK_GW_HTTPSERVEROPTIONS_CERTIFICATES", "[{\"domain_name\":\"testCerts\"}]")
+	t.Setenv("TYK_GW_HTTPSERVEROPTIONS_CERTIFICATES", "[{\"domain_name\":\"testCerts\"}]")
 	err := envconfig.Process("TYK_GW", &c)
 	if err != nil {
 		t.Fatal(err)
 	}
 
 	assert.Len(t, c.HttpServerOptions.Certificates, 1, "TYK_GW_HTTPSERVEROPTIONS_CERTIFICATES should have len 1")
 	assert.Equal(t, "testCerts", c.HttpServerOptions.Certificates[0].Name, "TYK_GW_HTTPSERVEROPTIONS_CERTIFICATES domain_name should be equals to testCerts")
+}
+
+// TestSecretsDecoder tests env variable decoding for TYK_GW_SECRETS.
+// It confirms that key pairs should be provided as a comma separated
+// list of keys and values, additionally separated by `:` (colon).
+func TestSecretsDecoder(t *testing.T) {
+	var c Config
+	t.Setenv("TYK_GW_SECRETS", "key:value,key2:value2")
```

Comment:
The added test case confirms the delimiter as `:` and not `=`; via [envconfig](https://github.com/kelseyhightower/envconfig).